presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -197,18 +195,10 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Set<VariableReferenceExp
         {
             Set<VariableReferenceExpression> expectedFilterInputs = new HashSet<>();
             if (node.getFilter().isPresent()) {
-                if (isExpression(node.getFilter().get())) {
-                    expectedFilterInputs = ImmutableSet.<VariableReferenceExpression>builder()
-                            .addAll(VariablesExtractor.extractUnique(castToExpression(node.getFilter().get()), TypeProvider.viewOf(variableAllocator.getVariables())))
-                            .addAll(context.get())
-                            .build();
-                }
-                else {
-                    expectedFilterInputs = ImmutableSet.<VariableReferenceExpression>builder()
-                            .addAll(VariablesExtractor.extractUnique(node.getFilter().get()))
-                            .addAll(context.get())
-                            .build();
-                }
+                expectedFilterInputs = ImmutableSet.<VariableReferenceExpression>builder()
+                        .addAll(VariablesExtractor.extractUnique(node.getFilter().get()))
```

Comment:
There are many places where we can use static import, this is in a way old codebase, so I would leave it to you if you want to fix it.